services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/file/FileHandleManagerImpl.java

```diff
@@ -445,12 +451,10 @@ private String getCloudFrontSignedUrlForS3FileHandle(S3FileHandle handle) {
 		String creationDate = AWS4SignerUtils.formatTimestamp(creationTimeMS);
 		Date expirationDate = new Date(creationTimeMS + PRESIGNED_URL_EXPIRE_TIME_MS);
 
-		String resourceUrl;
-		try {
-		URIBuilder uriBuilder = new URIBuilder()
-				.setScheme("https")
-				.setHost(distributionDomainName)
-				.setPath(handle.getKey());
+		UriComponentsBuilder uriBuilder = UriComponentsBuilder.newInstance()
+				.scheme("https")
+				.host(distributionDomainName)
+				.path(SdkHttpUtils.urlEncode(handle.getKey(), true));
```

Comment:
Maybe let us put this in variables for readability, including the boolean value. Also add a comment to explain why we are doing this.